discourseparsing/tune_rst_parser.py

```diff
@@ -1,49 +1,73 @@
-import json
-from configparser import ConfigParser
+from collections import Counter
 from concurrent.futures import ProcessPoolExecutor
+from configparser import ConfigParser
 from functools import partial
+from os.path import abspath, exists, join
 
 import numpy as np
+from nltk.tree import ParentedTree
 from skll.experiments import run_configuration
 from skll.learner import Learner
-from nltk.tree import ParentedTree
-
-from discourseparsing.discourse_parsing import Parser
-from discourseparsing.extract_actions_from_trees import extract_parse_actions
-from discourseparsing.collapse_rst_labels import collapse_rst_labels
-from discourseparsing.rst_eval import predict_and_evaluate_rst_trees
-
-
-def train_rst_parsing_model(working_path, model_path, parameter_settings):
-    '''
-    parameter_settings is a dict of scikit-learn hyperparameter settings
-    '''
-
-    C_value = parameter_settings['C']
-    working_subdir = os.path.join(working_path, 'C{}'.format(C_value))
-    assert not os.path.exists(working_subdir)
-    os.makedirs(working_subdir)
-
-    if not os.path.exists(model_path):
-        os.makedirs(model_path)
 
+from .collapse_rst_labels import collapse_rst_labels
+from .discourse_parsing import Parser
+from .extract_actions_from_trees import extract_parse_actions
+from .rst_eval import predict_and_evaluate_rst_trees
+
+
+def train_rst_parsing_model(working_path, model_path, C):
+    """
+    Train an RST parsing model on pre-extracted featuers and save to disk.
+
+    This function trains a logistic regression RST parsing model on features
+    that are assumed to have already been extracted to a file called
+    ``rst_parsing.jsonlines`` under ``working_path``. The  ``C``
+    hyperparameter of the model is set to the value provided as input
+    and the model is saved under ``model_path`` with the name
+    ``rst_parsing_all_feats_LogisticRegression.model``.
+
+    Parameters
+    ----------
+    working_path : str
+        Path to the directory where the pre-extracted SKLL training features
+        are stored.
+    model_path : str
+        Path to the directory where the trained model will be saved.
+    C : float
+        The value for the ``C`` hyperparameter value to be used
+        when training the model.
+    """
+    # create a sub-directory under ``working_path`` to store the logs
+    working_subdir = join(working_path, f"C{C}")
+    os.makedirs(working_subdir, exist_ok=False)
+
+    # create ``model_path`` unless it already exists
+    os.makedirs(model_path, exist_ok=True)
+
+    # set up the learner name and settings for use in the SKLL configuration
     learner_name = 'LogisticRegression'
-    fixed_parameters = [{'random_state': 123456789, 'penalty': 'l1',
-                         'C': C_value}]
+    fixed_parameters = [{"random_state": 123456789,
+                         "penalty": 'l1',
+                         'C': C}]
 
-    # Make the SKLL config file.
+    # create the SKLL config dictionary
```

Comment:
`config dictionary` should be `config file`. It is making a file under `working_subdir`

Reply:
Well, we are creating a dictionary and then writing it to a file.